### PR TITLE
feat(server): add `btmRecommendationModelChanges` query for client recommendation collection

### PR DESCRIPTION
**Pull request by overbit**
Close: #7818

## PR Details

Expose a new gql Query for collecting BTM recommendation resource.

Documentation has been added to the gql schema itself to simplify integration. Check `btmRecommendationModelChanges` in `server` playground or apollo studio.

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/ai/ai.service.ts [hunk 1]
```diff
@@ -222,7 +227,7 @@ export class AiService {
               name: resource.name,
```
Comment (morhag90) [inline packages/amplication-server/src/core/ai/ai.service.ts:227]: Why do you need the where filter for the resource entities?

File: packages/amplication-server/src/core/ai/btm-manager.service.ts [hunk 2]
```diff
@@ -56,17 +53,25 @@ export class BtmManagerService {
...
+              const entityNameIdMap = originalResource.entities.reduce(
+                (map, entity) => {
+                  map[entity.name] = entity;
+                  return map;
+                },
+                {} as Record<string, EntityPartial>
+              );
+
+              return {
+                id: undefined,
```
Comment (morhag90) [inline packages/amplication-server/src/core/ai/btm-manager.service.ts:66]: Is this the tempResourceId field? If it is, Why not generate a tempId like a random guid?

File: packages/amplication-server/src/core/ai/ai.service.ts (hunk 1 above)
Comment (overbit) [reply packages/amplication-server/src/core/ai/ai.service.ts:227]: Where do you mean?

File: packages/amplication-server/src/core/ai/dto/btm-recommendation-model-changes-args.dto.ts
Comment (abrl91) [inline packages/amplication-server/src/core/ai/dto/btm-recommendation-model-changes-args.dto.ts:1]: The name of the file is different from what we are doing. Even though what you did is the recommended practice, I think we should be consistent (same for other DTOs file names)

Comment (overbit) [reply packages/amplication-server/src/core/ai/dto/btm-recommendation-model-changes-args.dto.ts:1]: handled here https://github.com/amplication/amplication/pull/7842

**Review by abrl91 — APPROVED** [review]

**Pull request merged**